elastica/dissipation.py

```diff
@@ -220,22 +223,90 @@ def __init__(self, filter_order: int, **kwargs):
                 "Invalid filter order! Filter order must be a positive integer."
             )
         self.filter_order = filter_order
-        self.velocity_filter_term = np.zeros_like(self._system.velocity_collection)
-        self.omega_filter_term = np.zeros_like(self._system.omega_collection)
+
+        if self._system.ring_rod_flag:
+            # There are two periodic boundaries
+            blocksize = self._system.n_elems + 2
+            self.velocity_filter_term = np.zeros((3, blocksize))
+            self.omega_filter_term = np.zeros((3, blocksize))
+            self.filter_function = (
+                _filter_function_generator_periodic_condition_ring_rod
+            )
+
+        else:
+            self.velocity_filter_term = np.zeros_like(self._system.velocity_collection)
+            self.omega_filter_term = np.zeros_like(self._system.omega_collection)
+            self.filter_function = _filter_function_generator_periodic_condition
 
     def dampen_rates(self, rod: RodType, time: float) -> None:
-        nb_filter_rate(
-            rate_collection=rod.velocity_collection,
-            filter_term=self.velocity_filter_term,
-            filter_order=self.filter_order,
-        )
-        nb_filter_rate(
-            rate_collection=rod.omega_collection,
-            filter_term=self.omega_filter_term,
-            filter_order=self.filter_order,
+
+        self.filter_function(
+            rod.velocity_collection,
+            self.velocity_filter_term,
+            rod.omega_collection,
+            self.omega_filter_term,
+            self.filter_order,
         )
 
 
+@njit(cache=True)
+def _filter_function_generator_periodic_condition_ring_rod(
+    velocity_collection,
```

Comment:
I'm not sure if these functions are generators as in they aren't returning anything? maybe just remove `generator` from the name?